arch/arm/core/fault.c

```diff
@@ -640,11 +662,24 @@ static u32_t _FaultDump(const NANO_ESF *esf, int fault)
  * Note: exc_return argument shall only be used by the Fault handler if we are
  * building Secure Firmware.
  */
-void _Fault(const NANO_ESF *esf, u32_t exc_return)
+void _Fault(NANO_ESF *esf, u32_t exc_return)
 {
 	u32_t reason = _NANO_ERR_HW_EXCEPTION;
 	int fault = SCB->ICSR & SCB_ICSR_VECTACTIVE_Msk;
 
+#ifdef CONFIG_USERSPACE
+	for (int i = 0; i < ARRAY_SIZE(exceptions); i++) {
+		/* Mask out instruction mode */
+		u32_t start = (u32_t)exceptions[i].start & ~0x1;
+		u32_t end = (u32_t)exceptions[i].end & ~0x1;
+
+		if (esf->pc >= start && esf->pc < end) {
+			esf->pc = (u32_t)(exceptions[i].fixup);
+			return;
```

Comment:
As I understand, this code block will make `_Fault()` to by-pass the standard fault handling process handling. So we need to **manually** perform some actions that are now incorporated into the relative fault handlers (e.g. clear fault status register sticky bits, reset the fault address register etc.)

Reply:
> by-pass the standard fault handling process handling

Where does this take place? I cannot find it. All _Fault seems to do is call various code to dump out information and then call SysFatalErrorHandler() which then returns.

Reply:
In line 678 we seem to return from _Fault(). So, for example, if this was a BusFault, we are not going to clear the Sticky bits.

Reply:
The fault handlers do not only dump information, but also clear fault status.

Reply:
I am suggesting to refactor and clean up this code in Fault.c,so fault dumping and fault status clearing are somewhat separated....

Reply:
Well, this is a bad situation because in this scenario I really don't want anything printed.
Not only that, but this code doesn't even run if CONFIG_FAULT_DUMP isn't set. And I have to propagate back to the main _Fault() somehow that we need to return without calling SysFatalErrorHandler.
Do you have any advice other than rewriting fault.c for ARM?